futures/src/channel.rs

```diff
@@ -145,73 +145,104 @@ pub struct ChannelFlowOptions {
 
 impl<T: AsyncRead+AsyncWrite+Send+'static> Channel<T> {
     /// create a channel
-    pub fn create(transport: Arc<Mutex<AMQPTransport<T>>>) -> Box<Future<Item = Self, Error = io::Error> + Send + 'static> {
+    pub fn create(transport: Arc<Mutex<AMQPTransport<T>>>) -> impl Future<Item = Self, Error = io::Error> + Send {
         let channel_transport = transport.clone();
         let create_channel = future::poll_fn(move || {
-            let mut transport = try_lock_transport!(channel_transport);
```

Comment:
why are you reverting that?

Reply:
It's not the only place where you reverted stuff btw, especially this pattern.
And please amend the fixes to your commit

Reply:
Could you please point me the other places? I couldn't find them reading diff.

Reply:
It appears you fixed it since the review